.travis.yml

```diff
@@ -10,16 +9,37 @@ php:
   - hhvm
  
 matrix:
-    allow_failures:
-        - php: hhvm
+  fast_finish: true
+  include:
+    - php: 5.3
+      env: SYMFONY_VERSION='2.3.*' FOSUSERBUNDLE_VERSION='~1.3'
+    - php: 5.3
+      env: SYMFONY_VERSION='2.7.*' FOSUSERBUNDLE_VERSION='2.0.*@dev'
+    - php: 5.6
+      env: SYMFONY_VERSION='2.3.*' FOSUSERBUNDLE_VERSION='~1.3'
+    - php: 5.6
+      env: SYMFONY_VERSION='2.7.*' FOSUSERBUNDLE_VERSION='2.0.*@dev'
+    - php: 5.6
+      env: SYMFONY_VERSION=2.8.*@dev FOSUSERBUNDLE_VERSION='2.0.*@dev'
+    - php: 5.6
+      env: SYMFONY_VERSION="3.0.x-dev as 2.8" FOSUSERBUNDLE_VERSION='2.0.*@dev'
+  allow_failures:
+    - php: hhvm
+    - env: SYMFONY_VERSION=2.8.*@dev
+    - env: SYMFONY_VERSION="3.0.x-dev as 2.8"
 
-env:
-  - SYMFONY_VERSION='2.3.*' FOSUSERBUNDLE_VERSION='~1.2'
-  - SYMFONY_VERSION='2.4.*' FOSUSERBUNDLE_VERSION='~1.2'
-  - SYMFONY_VERSION='2.3.*' FOSUSERBUNDLE_VERSION='2.0.*@dev'
-  - SYMFONY_VERSION='2.4.*' FOSUSERBUNDLE_VERSION='2.0.*@dev'
+sudo: false
 
-before_script:
-    - composer require --no-update symfony/symfony=${SYMFONY_VERSION}
-    - composer require --no-update friendsofsymfony/user-bundle=${FOSUSERBUNDLE_VERSION}
-    - composer install --prefer-source
+cache:
+  directories:
+    - $HOME/.composer/cache
```

Comment:
you should persist only the ``$HOME/.composer/cache/files``. This will avoid making Travis update the stored cache for each build because of changes in the metadata downloaded from packagist and cache in ``.composer/cache/repo``). Persisting the Packagist downloaded files between builds is not very useful as they will probably have changed between builds (the packages you depend on are active packages for most of them)